flow/project.py

```diff
@@ -3796,14 +4096,21 @@ def _select_jobs_from_args(self, args):
                 "Cannot provide both -j/--job-id and -f/--filter or --doc-filter in combination.")
 
         if args.job_id:
-            try:
-                return [self.open_job(id=job_id) for job_id in args.job_id]
-            except KeyError as error:
-                raise LookupError("Did not find job with id {}.".format(error))
-        else:
+            aggregate = set()  # set in order to avoid duplicate ids
+            for id in args.job_id:
+                # TODO: We need to add support for aggregation id parameter
+                # for the -j flag ('agg-...')
+                try:
+                    aggregate.add((self.open_job(id=id),))
+                except KeyError as error:
+                    raise LookupError("Did not find job with id {}.".format(error))
+            return list(aggregate)
+        elif args.filter or args.doc_filter:
             filter_ = parse_filter_arg(args.filter)
             doc_filter = parse_filter_arg(args.doc_filter)
             return JobsCursor(self, filter_, doc_filter)
+        else:
+            return None
```

Comment:
The primary reason was to cover an edge case where users pass in an empty list in the `jobs` parameter. `None` here means that allow all the aggregate operation pairs to get executed. But we use an empty list instead of None then the behaviour would be unwanted as the user passed in no jobs. 

Please suggest what should I do here.

Reply:
I see, so the reason that this is confusing is because `JobsCursor` internally takes advantage of the fact `Project._find_job_ids` correctly handles `None` values for `filter` and `doc_filter`. You can see this because the constructor for `JobsCursor` does not provide default arguments for `filter` and `doc_filter`, so it just stores `None` and then passes it to `_find_job_ids` in the `__iter__` method.

I think the correct solution here is probably to generalize the `JobsCursor` into an `AggregateCursor` class that is capable of the same types of lazy iteration, but over jobs _or_ aggregates rather than just jobs. @b-butler thoughts?

Reply:
@bdice why 🤔

Reply:
I meant to follow up on this with a comment. I had trouble figuring out why the return value should be `None`. Also I was not excited about creating another layer of complexity in aggregation with a Cursor-like class, but could not come up with a better alternative.

Reply:
So basically, `None` here means that no specific preference is made by the user for selecting jobs or aggregates. I believe, if not intuitive, this is one of the least complex approaches to the problem I was trying to solve.

Reply:
I think a cursor like object that can be iterated over multiple times is what is needed. This can wait for now though.